random/README.md

```diff
@@ -0,0 +1,181 @@
+# Random - Generating Random Numbers
+
+This example shows you how to use the Blocky Attestation Service (Blocky AS) to
+attest simple function call that generates random numbers to simulate a die roll.
+
+Before starting this example, make sure you are familiar with the
+[Hello World - Attesting a Function Call](../hello_world_attest_fn_call/README.md)
+and the
+[Error Handling](../error_handling/README.md)
+examples.
+
+You'll learn how to:
+
+- Generate random numbers in the Blocky AS runtime
+- Create a function that returns a random number within a given range
+
+*Note: Random number generation only works on Blocky AS versions 0.1.0-beta.5
+and above. While this example may run on previous versions it will generate
+deterministic number streams.*
+
+## Setup
+
+- Install the Blocky AS CLI by following the
+  [setup instructions](https://blocky-docs.redocly.app/attestation-service/setup)
+  in the Blocky AS documentation.
+- Make sure you also have
+  [Docker](https://www.docker.com/) and [jq](https://jqlang.org/) installed on
+  your system.
+
+## Quick Start
+
+To run this example, call:
+
+```bash
+make run
+```
+
+You will see the following output extracted from a Blocky AS response:
+
+```
+Output:
+{
+  "Success": true,
+  "Error": "",
+  "Value": 3
+}
+```
+
+## Walkthrough
+
+### Step 1: Create a parameterized random function
+
+We'll demonstrate random by emulating a die roll in `rollDie` in
+[`main.go`](./main.go). As in previous examples, we will call this function
+using the `bky-as` CLI by passing in the
+[`fn-call.json`](./fn-call.json) file contents:
+
+```json
+{
+  "code_file": "tmp/x.wasm",
+  "function": "rollDie",
+  "input": {
+    "sides": 20
+  }
+}
+```
+
+Notice the `input` section, which contains the parameters for `rollDie`,
+specifically the `sides` field set to `20`. Meaning, `rollDie` will return
+a number in the range [1,20].
+
+Next, we define the `rollDie` function in [`main.go`](./main.go):
+
+```go
+type Args struct {
+    Sides uint64 `json:"sides"`
+}
+
+//export rollDie
+func rollDie(inputPtr uint64, secretPtr uint64) uint64 {
```

Comment:
I think this is out of date with respect to main.go. You might be able to ignore the next couple of comments.

Reply:
Ah good catch. Forgot to update the code here after changing in `main.go`. Fixed.